edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionMetadata.cs

```diff
@@ -18,7 +18,7 @@ public ConnectionMetadata(string productInfo, string modelId, string edgeProduct
 
         public ConnectionMetadata(string productInfo, Option<string> modelId, string edgeProductInfo)
         {
-            Preconditions.CheckNonWhiteSpace(edgeProductInfo, nameof(edgeProductInfo));
```

Comment:
You said you didn't want to cause breaking changes, right? Why can't we change the tests to send a edgeProductInfo?

Reply:
Are you concerned about breaking changes from 1.0.10 RC to 1.0.10?

Reply:
The breaking change I'm concerned about is that the old ProductInfo allows Empty string/whitespace as edge product info, but the new MetadataStore does not. This change relaxes it back to how the old ProductInfoStore had it.
So it's breaking change between 1.0.10 RC-1 -> 1.0.10 RC-2 and above

Reply:
The breaking change I'm concerned about is that the old ProductInfo allows Empty string/whitespace as edge product info, but the new MetadataStore does not. This change relaxes it back to how the old ProductInfoStore had it.
So it's breaking change between 1.0.10 RC-1 -> 1.0.10 RC-2 and above